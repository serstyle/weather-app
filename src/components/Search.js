import React from 'react'

class Search extends React.Component{
	render(){return(
		<div>
			<form onSubmit={this.props.handleSubmit}>
				<div className="row">
			        <div className="input-field col s12">
						<label>City</label>
						<input aria-label='search city' onChange={this.props.handleChange}type='text' />
						<button className='waves-effect waves-light btn'>Submit</button>
						{this.props.error?<span style={{marginLeft:'10px'}} className="helper-text red-text">didn't find the city</span>: null}
					</div>
				</div>	
			</form>
		</div>
		)}
}				

export default Search;


// in class bc i wanted to try componentdidcatch


// const Search = ({handleChange, handleSubmit	}) =>{
// 	return(
// 		<div>
// 			<form onSubmit={handleSubmit}>
// 				<label>City : </label>
// 				<input onChange={handleChange}type='text' />
// 				<button>Submit</button>
// 			</form>
// 		</div>
// 		)
// }			