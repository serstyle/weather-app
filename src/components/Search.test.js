import React from 'react'
import {shallow} from 'enzyme'
import Search from './Search'



describe('should snapshot component', () => {
    it('snapshot with this.props.error true', () => {
        let wrap = shallow(<Search error={true}/>)
        expect(wrap).toMatchSnapshot()
    })
    it('snapshot with this.props.error false', () => {
        let wrap = shallow(<Search error={false}/>)
        expect(wrap).toMatchSnapshot()
    })
})

describe('handle change or submit', ()=>{
    const mockHandleChange = jest.fn()
    const mockHandleSubmit = jest.fn()
    let wrap = shallow(<Search handleChange={mockHandleChange} handleSubmit={mockHandleSubmit}/>)
    it('should handleChange', () => {
      wrap.find('input').simulate('change')
      wrap.find('input').simulate('change')
      wrap.find('input').simulate('change')
      wrap.find('input').simulate('change')
      expect(mockHandleChange).toHaveBeenCalledTimes(4)
    })
    it('should handleSubmit', () => {
        wrap.find('form').simulate('submit')
        expect(mockHandleSubmit).toHaveBeenCalledTimes(1)
      })
    
})



