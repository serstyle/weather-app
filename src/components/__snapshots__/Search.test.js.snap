// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot component snapshot with this.props.error false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    error={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <label>
              City
            </label>
            <input
              aria-label="search city"
              type="text"
            />
            <button
              className="waves-effect waves-light btn"
            >
              Submit
            </button>
          </div>
        </div>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <label>
              City
            </label>
            <input
              aria-label="search city"
              type="text"
            />
            <button
              className="waves-effect waves-light btn"
            >
              Submit
            </button>
          </div>
        </div>,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="input-field col s12"
          >
            <label>
              City
            </label>
            <input
              aria-label="search city"
              type="text"
            />
            <button
              className="waves-effect waves-light btn"
            >
              Submit
            </button>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                City
              </label>,
              <input
                aria-label="search city"
                type="text"
              />,
              <button
                className="waves-effect waves-light btn"
              >
                Submit
              </button>,
              null,
            ],
            "className": "input-field col s12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "City",
              },
              "ref": null,
              "rendered": "City",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "search city",
                "onChange": undefined,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "waves-effect waves-light btn",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
            null,
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <div
            className="row"
          >
            <div
              className="input-field col s12"
            >
              <label>
                City
              </label>
              <input
                aria-label="search city"
                type="text"
              />
              <button
                className="waves-effect waves-light btn"
              >
                Submit
              </button>
            </div>
          </div>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="input-field col s12"
            >
              <label>
                City
              </label>
              <input
                aria-label="search city"
                type="text"
              />
              <button
                className="waves-effect waves-light btn"
              >
                Submit
              </button>
            </div>
          </div>,
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="input-field col s12"
            >
              <label>
                City
              </label>
              <input
                aria-label="search city"
                type="text"
              />
              <button
                className="waves-effect waves-light btn"
              >
                Submit
              </button>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  City
                </label>,
                <input
                  aria-label="search city"
                  type="text"
                />,
                <button
                  className="waves-effect waves-light btn"
                >
                  Submit
                </button>,
                null,
              ],
              "className": "input-field col s12",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "City",
                },
                "ref": null,
                "rendered": "City",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "search city",
                  "onChange": undefined,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "className": "waves-effect waves-light btn",
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
              null,
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`should snapshot component snapshot with this.props.error true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    error={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <label>
              City
            </label>
            <input
              aria-label="search city"
              type="text"
            />
            <button
              className="waves-effect waves-light btn"
            >
              Submit
            </button>
            <span
              className="helper-text red-text"
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            >
              didn't find the city
            </span>
          </div>
        </div>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <label>
              City
            </label>
            <input
              aria-label="search city"
              type="text"
            />
            <button
              className="waves-effect waves-light btn"
            >
              Submit
            </button>
            <span
              className="helper-text red-text"
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            >
              didn't find the city
            </span>
          </div>
        </div>,
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="input-field col s12"
          >
            <label>
              City
            </label>
            <input
              aria-label="search city"
              type="text"
            />
            <button
              className="waves-effect waves-light btn"
            >
              Submit
            </button>
            <span
              className="helper-text red-text"
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            >
              didn't find the city
            </span>
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                City
              </label>,
              <input
                aria-label="search city"
                type="text"
              />,
              <button
                className="waves-effect waves-light btn"
              >
                Submit
              </button>,
              <span
                className="helper-text red-text"
                style={
                  Object {
                    "marginLeft": "10px",
                  }
                }
              >
                didn't find the city
              </span>,
            ],
            "className": "input-field col s12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "City",
              },
              "ref": null,
              "rendered": "City",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "search city",
                "onChange": undefined,
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "waves-effect waves-light btn",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "didn't find the city",
                "className": "helper-text red-text",
                "style": Object {
                  "marginLeft": "10px",
                },
              },
              "ref": null,
              "rendered": "didn't find the city",
              "type": "span",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <div
            className="row"
          >
            <div
              className="input-field col s12"
            >
              <label>
                City
              </label>
              <input
                aria-label="search city"
                type="text"
              />
              <button
                className="waves-effect waves-light btn"
              >
                Submit
              </button>
              <span
                className="helper-text red-text"
                style={
                  Object {
                    "marginLeft": "10px",
                  }
                }
              >
                didn't find the city
              </span>
            </div>
          </div>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="input-field col s12"
            >
              <label>
                City
              </label>
              <input
                aria-label="search city"
                type="text"
              />
              <button
                className="waves-effect waves-light btn"
              >
                Submit
              </button>
              <span
                className="helper-text red-text"
                style={
                  Object {
                    "marginLeft": "10px",
                  }
                }
              >
                didn't find the city
              </span>
            </div>
          </div>,
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="input-field col s12"
            >
              <label>
                City
              </label>
              <input
                aria-label="search city"
                type="text"
              />
              <button
                className="waves-effect waves-light btn"
              >
                Submit
              </button>
              <span
                className="helper-text red-text"
                style={
                  Object {
                    "marginLeft": "10px",
                  }
                }
              >
                didn't find the city
              </span>
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label>
                  City
                </label>,
                <input
                  aria-label="search city"
                  type="text"
                />,
                <button
                  className="waves-effect waves-light btn"
                >
                  Submit
                </button>,
                <span
                  className="helper-text red-text"
                  style={
                    Object {
                      "marginLeft": "10px",
                    }
                  }
                >
                  didn't find the city
                </span>,
              ],
              "className": "input-field col s12",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "City",
                },
                "ref": null,
                "rendered": "City",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-label": "search city",
                  "onChange": undefined,
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit",
                  "className": "waves-effect waves-light btn",
                },
                "ref": null,
                "rendered": "Submit",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "didn't find the city",
                  "className": "helper-text red-text",
                  "style": Object {
                    "marginLeft": "10px",
                  },
                },
                "ref": null,
                "rendered": "didn't find the city",
                "type": "span",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
